### PR TITLE
CustomSelectControlV2: Handle long strings in selected value

### PR DESCRIPTION
**Pull request by mirka**
Part of #55023

## What?

Correctly truncate and style long strings in the currently selected value in CustomSelectControlV2.

## Why?

Long strings were not styled correctly.

### Before

Long strings look wrong

## Testing Instructions

Compare the "With Long Labels" stories between CustomSelectControl (v1) and CustomSelectControlV2 Legacy. All `size` variants should look correct.

> [!NOTE]  
> You may notice that the `LABEL` is shifted down 1px in the v1. I looked into it and v1 is actually wrong in this case, so let's ignore. (FWIW, it will look correct in wp-admin due to a style in the `forms.css` stylesheet.)

## Screenshots or screencast 

|| v1 | v2 Default & Legacy |
|--------|--------|-------|
|Long labels|long v1|long v2 legacy|
|With hints|hint v1|hint v2 legacy|

### CODE REVIEW

File: packages/components/src/custom-select-control-v2/custom-select.tsx [hunk 1]
```diff
@@ -74,7 +74,7 @@ const CustomSelectButton = ( {
 			// move selection rather than open the popover
 			showOnKeyDown={ false }
 		>
-			<div>{ computedRenderSelectedValue( currentValue ) }</div>
```
Comment (mirka) [inline packages/components/src/custom-select-control-v2/custom-select.tsx:77]: `div` is unnecessary.

File: packages/components/src/custom-select-control-v2/legacy-component/index.tsx [hunk 2]
```diff
@@ -82,12 +82,12 @@ function CustomSelectControl( props: LegacyCustomSelectProps ) {
 		);
 
 		return (
-			<>
+			<Styled.SelectedExperimentalHintWrapper>
```
Comment (mirka) [inline packages/components/src/custom-select-control-v2/legacy-component/index.tsx:93]: Added a wrapper div to apply text truncation styles as in v1.

File: packages/components/src/custom-select-control-v2/styles.ts [hunk 3]
```diff
@@ -55,40 +66,41 @@ export const Select = styled( Ariakit.Select, {
...
 			},
 		};
 
 		return sizes[ size ] || sizes.default;
 	};
 
 	return css`
-		display: flex;
-		align-items: center;
-		justify-content: space-between;
```
Comment (mirka) [inline packages/components/src/custom-select-control-v2/styles.ts:80]: No reason for this to be a flexbox.

File: packages/components/src/custom-select-control-v2/styles.ts [hunk 4]
```diff
@@ -55,40 +66,41 @@ export const Select = styled( Ariakit.Select, {
...
+		display: block;
 		background-color: ${ COLORS.theme.background };
 		border: none;
+		color: ${ COLORS.theme.foreground };
 		cursor: pointer;
+		font-family: inherit;
 		font-size: ${ CONFIG.fontSize };
+		text-align: left;
```
Comment (mirka) [inline packages/components/src/custom-select-control-v2/styles.ts:95]: Text align was missing.

File: packages/components/src/custom-select-control-v2/styles.ts [hunk 5]
```diff
@@ -55,40 +66,41 @@ export const Select = styled( Ariakit.Select, {
...
 	return css`
-		display: flex;
-		align-items: center;
-		justify-content: space-between;
+		display: block;
 		background-color: ${ COLORS.theme.background };
 		border: none;
+		color: ${ COLORS.theme.foreground };
 		cursor: pointer;
+		font-family: inherit;
```
Comment (mirka) [inline packages/components/src/custom-select-control-v2/styles.ts:93]: Correct `color` and `font-family` was missing.

File: packages/components/src/custom-select-control-v2/styles.ts [hunk 6]
```diff
@@ -55,40 +66,41 @@ export const Select = styled( Ariakit.Select, {
...
 		font-size: ${ CONFIG.fontSize };
+		text-align: left;
 		width: 100%;
 
 		&[data-focus-visible] {
 			outline: none; // handled by InputBase component
 		}
 
 		${ getSize() }
+		${ ! hasCustomRenderProp && truncateStyles }
```
Comment (mirka) [inline packages/components/src/custom-select-control-v2/styles.ts:103]: We shouldn't add truncate styles when the consumer passes a custom render function.

File: packages/components/src/custom-select-control-v2/styles.ts [hunk 7]
```diff
@@ -55,40 +66,41 @@ export const Select = styled( Ariakit.Select, {
 		const sizes = {
 			compact: {
 				[ heightProperty ]: 32,
-				paddingInlineStart: space( 2 ),
-				paddingInlineEnd: space( 1 ),
+				paddingInlineStart: 8,
```
Comment (tyxla) [inline packages/components/src/custom-select-control-v2/styles.ts:69]: Is there a good reason to stop using the grid space util and hardcode the padding widths like that?

File: packages/components/src/custom-select-control-v2/styles.ts (hunk 4 above)
Comment (tyxla) [reply packages/components/src/custom-select-control-v2/styles.ts:95]: Nice catch 👍 I wonder how this didn't come up earlier.

**Review by tyxla — APPROVED** [review]
Working well in my testing and the code looks good 👍 

Thanks for cleaning up the styles while passing through!

File: packages/components/src/custom-select-control-v2/styles.ts (hunk 7 above)
Comment (fullofcaffeine) [reply packages/components/src/custom-select-control-v2/styles.ts:69]: Curious about that, too.

**Review by fullofcaffeine — APPROVED** [review]
LGTM! Thanks for working on this.

I started fixing other styling issues here: https://github.com/WordPress/gutenberg/pull/62424 (it's been split from [#](https://github.com/WordPress/gutenberg/pull/61272)), so I'll rebase from this, review the current issues, and continue working there.

File: packages/components/src/custom-select-control-v2/styles.ts (hunk 3 above)
Comment (ciampo) [reply packages/components/src/custom-select-control-v2/styles.ts:80]: Knowing past Marco, there's a chance flexbox was used to automatically trim extra white space (although it's also very possible that this is a remnant of a previous style and it's totally fine to remove it)

Comment (ciampo) [reply packages/components/src/custom-select-control-v2/styles.ts:80]: Probably totally fine to remove it :D

File: packages/components/src/custom-select-control-v2/styles.ts (hunk 7 above)
Comment (mirka) [reply packages/components/src/custom-select-control-v2/styles.ts:69]: No big reason other than that I thought it was simpler to read in context, especially when I want to add the `chevronIconSize` like this. (`space()` returns a `calc()`).

**Pull request merged**